soc/arm/st_stm32/stm32l4/power.c

```diff
@@ -27,82 +27,124 @@ LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
 #define RCC_STOP_WAKEUPCLOCK_SELECTED LL_RCC_STOP_WAKEUPCLOCK_HSI
 #endif
 
-/* Invoke Low Power/System Off specific Tasks */
-__weak void pm_state_set(enum pm_state state, uint8_t substate_id)
+void enter_ultra_low_power_mode(void)
 {
-	if (state != PM_STATE_SUSPEND_TO_IDLE) {
-		LOG_DBG("Unsupported power state %u", state);
-		return;
+	/* Configure CPU core */
+	/* Enable CPU deep sleep mode */
+	LL_LPM_EnableDeepSleep();
+	LL_DBGMCU_DisableDBGStandbyMode();
+	/* Enter ultra_low-power mode */
+
+	for (;;) {
```

Comment:
Is that actually better than just calling k_cpu_idle()?

Reply:
Sorry, I haven't been clear enough. The question was not about the loop, but about adding a separate wfi path instead of going through k_cpu_idle() which also does wfi.

Reply:
OK I'll take a look and test if this change works.

Reply:
I mean, why not just adapt the approach used in stm32u5/power.c? I think the code could be almost identical.